### ISSUE TITLE
hasOneDeepFromRelations returns collection

### ISSUE THREAD

**Issue by Felda01** [staudenmeir/laravel-adjacency-list#181]
Hello,

`latestBuildingModel` returns 1 model (`BuildingModel`), but `descendantLatestBuildingModel` returns collection (`BuildingModel`). Is `latestOfMany()` supported in `hasOneDeepFromRelations` ?

`Building` → descendants → `Building` → has many → `BuildingModel`

```
use Staudenmeir\LaravelAdjacencyList\Eloquent\HasRecursiveRelationships;
use Staudenmeir\EloquentHasManyDeep\HasRelationships;

class Building extends Model {

    use HasRecursiveRelationships;
    use HasRelationships;

    public function getParentKeyName()
    {
        return 'building_id';
    }

    public function building(): BelongsTo
    {
        return $this->belongsTo(Building::class, 'building_id', 'id');
    }

    public function buildings(): HasMany
    {
        return $this->hasMany(Building::class, 'building_id', 'id');
    }

    public function buildingModels(): HasMany
    {
        return $this->hasMany(BuildingModel::class, 'building_id', 'id');
    }

    public function latestBuildingModel(): HasOne
    {
        return $this->hasOne(BuildingModel::class)->latestOfMany();
    }

    public function descendantLatestBuildingModel(): HasOneDeep
    {
        return $this->hasOneDeepFromRelations(
            $this->descendants(),
            (new Building())->latestBuildingModel()
        );
    }
}
```

Thanks

**Comment by staudenmeir** [comment]
Hi @Felda01,
What version of the package are you using?

**Comment by Felda01** [comment]
Hi @staudenmeir,

`laravel/framework`: `v10.8.0`
`staudenmeir/eloquent-has-many-deep`: `v1.18.1`
`staudenmeir/laravel-adjacency-list`: `v1.13.6`

**Comment by staudenmeir** [comment]
This bug has been fixed in `v1.13.7`. Update the package with `composer update staudenmeir/laravel-adjacency-list`.

**Comment by Felda01** [comment]
Thanks so much :) and thanks for the awesome packages too!
